modules/multisignatures.js

```diff
@@ -80,158 +80,221 @@ function Multisignatures(cb, scope) {
 }
 
 // Public methods
+
 /**
- * Gets transaction from transaction id and add it to sequence and bus.
+ * Check if the provided signature is valid
  *
- * @param {Object} transaction - Contains transaction and signature
- * @param {function} cb - Callback function
- * @returns {setImmediateCallback} cb, err
- * @todo Add test coverage.
+ * @private
+ * @param {Object} signature - Signature data
+ * @param {string} [signature.publicKey] - Public key of account that created the signature (optional)
+ * @param {string} signature.transactionId - Id of transaction that signature was created for
+ * @param {string} signature.signature - Actual signature
+ * @param {Array} members - Array of multisignature account members - public keys
+ * @param {Object} transaction - Corresponding transaction grabbed from transaction pool
+ * @returns {boolean} isValid - true if signature passed verification, false otherwise
  */
-Multisignatures.prototype.processSignature = function(transaction, cb) {
-	if (!transaction) {
-		return setImmediate(
-			cb,
-			'Unable to process signature. Signature is undefined.'
-		);
-	}
-	const multisignatureTransaction = modules.transactions.getMultisignatureTransaction(
-		transaction.transactionId
-	);
-
-	function done(cb) {
-		library.balancesSequence.add(cb => {
-			const multisignatureTransaction = modules.transactions.getMultisignatureTransaction(
-				transaction.transactionId
-			);
-
-			if (!multisignatureTransaction) {
-				return setImmediate(cb, 'Transaction not found');
+__private.isValidSignature = (signature, members, transaction) => {
+	let isValid = false;
+
+	try {
+		// If publicKey is provided we can perform direct verify
+		if (signature.publicKey) {
+			// Check if publicKey is present as member of multisignature account in transaction
+			if (!members.includes(signature.publicKey)) {
+				library.logger.error(
+					'Unable to process signature, signer not in keysgroup.',
+					{ signature, members, transaction }
+				);
+				return isValid;
```

Comment:
This is confusing to return `isValid` here, always have to look up what value was assigned to it. It would be more clear just return `false` from this check. The only return statement in the end is more appropriate.